TackleTabby/Assets/Scripts/UI/ToggleHighlight.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections;
+using TMPro;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UIElements;
+public class ToggleHighlight : MonoBehaviour , IOverlayMenu
+{
+    [SerializeField]
+    private Canvas CatalogueCanvas;
+    [SerializeField] 
+    private Canvas HungerBarCanvas;
+
+    [SerializeField] 
+    private GameObject FieldGrayout;
+    [SerializeField] 
+    private GameObject CatalogueHighlight;
+    [SerializeField]
+    private GameObject HungerBarHighlight;
+    
+    [SerializeField]
+    private float waitingTime = 0.5f;
+    
+    public UnityEvent<IOverlayMenu> OnOpened;
+    public UnityEvent<IOverlayMenu> OnClosed;
+
+    private GameObject _currentHighlight;
+    private Canvas _currentCanvas;
+    private bool _mayCheck = true;
+    private bool _alreadyCaughtFish;
+    private bool _alreadyCaughtTrash;
+    private bool _isOpen;
+    
+    public void HasClosed()
+    {
+        _mayCheck = true;
+    }
+
+    public void CheckCaught(bool caughtAFish)
+    {
+        if (!_mayCheck)
+            return;
+        if (caughtAFish && !_alreadyCaughtFish)
+        {
+            _mayCheck = false;
+            _alreadyCaughtFish = true;
+            _currentCanvas = CatalogueCanvas;
+            _currentHighlight = CatalogueHighlight;
+            OpenOverlay();
+            return;
+        }
+        if (_alreadyCaughtTrash)
+            return;
+        _mayCheck = false;
```

Comment:
Perhaps leave some space between the if-statements for readability?
```suggestion
        if (!_mayCheck)
            return;

        if (caughtAFish && !_alreadyCaughtFish)
        {
            _mayCheck = false;
            _alreadyCaughtFish = true;
            _currentCanvas = CatalogueCanvas;
            _currentHighlight = CatalogueHighlight;
            OpenOverlay();
            return;
        }

        if (_alreadyCaughtTrash)
            return;

        _mayCheck = false;
```